core/src/main/java/org/apache/accumulo/core/clientImpl/bulk/LoadMappingIterator.java

```diff
@@ -72,12 +76,30 @@ public boolean hasNext() {
 
   @Override
   public Map.Entry<KeyExtent,Bulk.Files> next() {
-    Bulk.Mapping bm = gson.fromJson(reader, Bulk.Mapping.class);
-    if (renameMap != null) {
-      return new AbstractMap.SimpleEntry<>(bm.getKeyExtent(tableId),
-          bm.getFiles().mapNames(renameMap));
-    } else {
-      return new AbstractMap.SimpleEntry<>(bm.getKeyExtent(tableId), bm.getFiles());
+    try {
+      Bulk.Mapping bm = gson.fromJson(reader, Bulk.Mapping.class);
+      if (bm == null) {
+        throw new NoSuchElementException("No more elements in input");
+      }
+
+      KeyExtent currentKeyExtent = bm.getKeyExtent(tableId);
+
+      if (lastKeyExtent != null && currentKeyExtent.compareTo(lastKeyExtent) < 0) {
+        throw new IllegalStateException(
+            String.format("KeyExtents are not in sorted order: %s comes after %s", lastKeyExtent,
+                currentKeyExtent));
+      }
+
+      lastKeyExtent = currentKeyExtent;
+
+      if (renameMap != null) {
+        return new AbstractMap.SimpleEntry<>(currentKeyExtent, bm.getFiles().mapNames(renameMap));
+      } else {
+        return new AbstractMap.SimpleEntry<>(currentKeyExtent, bm.getFiles());
+      }
+
+    } catch (JsonSyntaxException | JsonIOException e) {
```

Comment:
These are both JsonParseException, so you only need to catch that parent. You could also shorten this try-catch block to just wrap the JSON parsing code, and not the entire method. Most of the rest of the logic can exist after the try-catch block.